solve.py

```diff
@@ -8,14 +8,28 @@ def load_words():
     return words
 
 
-def update_words(words, rule, required_chars, black_list_rules):
+def update_words(words, rule, required_chars, wrong_spot_chars, black_list_rules):
     updated = []
 
+    # Built exception rule
+    black_list_rule = [ _.replace('.', '') for _ in black_list_rules]
+    black_list_rule = ''.join(black_list_rule)
+
     for word in words:
-        if any([re.search(black_list_rule, word) for black_list_rule in black_list_rules]):
+
+        # Use regular expression to filter out the word that contain any blacklist characters
+        if re.search('['+black_list_rule+']', word) :
```

Comment:
Actually, this doesn't quite work or it has some assumption I'm not sure.
look at my change in this [commit](https://github.com/thanakritju/wordle-helper/commit/ee56501b7ca55d6af6ddfec34585086ff61fe9c4) I used to have `black_list_chars` to contain each chars
but it doesn't work for the input like `PROMO GGGBB` because **O** is both blacklisted and require char

Reply:
but my current implementation is quite wrong at the moment, yours might be more correct.